code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -41,7 +41,7 @@
 	. = ..()
 	//Cooldowns
 	if(ranged)
-		ranged_cooldown--
+		ranged_cooldown = max(0,ranged_cooldown--)
```

Comment:
Pretty sure this doesn't work. Because you're using post-decrement, you'll evaluate the `max()` with `ranged_cooldown`, decrement `ranged_cooldown`, then overwrite that with the return of `max()`.

As such, this'll never tick down.

You want to use `ranged_cooldown-1`.